### PR TITLE
Iteration 3 complete, tests passing

### PR DESCRIPTION
**Pull request by MacInnes**

### CODE REVIEW

**Comment by mikedao** [comment]
Good number of commits, great descriptive commit messages.

File: lib/translator.rb [hunk 1]
```diff
@@ -39,4 +39,39 @@ def initialize
                     "0" => "-----",
                     " " => " "}
   end
+
+  def eng_to_morse(input)
+    array = input.chars
+    translated_array = array.map do |each|
+      each = @dictionary[each.downcase]
+    end
+    return translated_array.join()
```
Comment (mikedao) [inline lib/translator.rb]: Good idea here, I'd have just iterated over the input.chars directly and then threw a join on it at the end like we did in class. Also, don't use () when its just empty. Avoid using the explicit return unless truly necessary.

File: lib/translator.rb [hunk 2]
```diff
@@ -39,4 +39,39 @@ def initialize
...
+  end
+
+  def from_file(filename)
+    file = File.new(filename, "r")
+    output = []
+    while (line = file.gets)
```
Comment (mikedao) [inline lib/translator.rb]: Every time you have a variable sitting outside of a looping structure and youre adding or doing something to it, it is a pattern that there is a better enumerable that you can use instead.

File: lib/translator.rb [hunk 3]
```diff
@@ -39,4 +39,39 @@ def initialize
...
+      output << line
+    end
+    eng_to_morse(output.join)
+  end
+
+  def morse_to_eng(input)
+    words = input.split("  ")
```
Comment (mikedao) [inline lib/translator.rb]: Good process, but this method is doing a bit too much - remember SRP - Single Responsibility Principle, a good rule of thumb is that a method shouldnt be more than 5-7 lines or so.

File: lib/translator.rb (hunk 1 above)
Comment (MacInnes) [reply lib/translator.rb]: gotcha - like input.chars.map( |e| e = @dictionary[e.downcase] }.join  ?  that's much cleaner.

File: lib/translator.rb (hunk 3 above)
Comment (MacInnes) [reply lib/translator.rb]: Wouldn't mind refactoring that with you sometime, I'm used to setting an output accumulator to gather all the data I want.

File: lib/translator.rb (hunk 1 above)
Comment (mikedao) [reply lib/translator.rb]: Yes, but still using the do/end syntax. I think that's much easier to read. Its not about cleanliness, it's about clarity.

**Pull request closed without merging**